packages/api/src/routes/batchRoutes.js

```diff
@@ -1,12 +1,33 @@
+import isAfter from 'date-fns/is_after';
+import addDays from 'date-fns/add_days';
 import { Router } from 'express';
 import { asyncHandler } from '../services/asyncHandler';
-import { getAllBatches } from '../db/collections/batch';
+import { getAllBatches, insertBatch } from '../db/collections/batch';
 
 const batchRoutes = Router();
 
-batchRoutes.post('/list', asyncHandler(async (req, res, next) => {
+batchRoutes.get('/list', asyncHandler(async (req, res, next) => {
   const batchList = await getAllBatches();
-  res.json({ batchList });
+  const { filter } = req.query;
+  if (filter === 'active') {
+    const activeBatches = batchList.filter((batch) => {
+      const date = new Date(batch.endDate);
+      return isAfter(date, Date.now());
+    });
+    res.json({ activeBatches });
+  } else {
+    res.json({ batchList });
+  }
+  next();
+}));
+
+batchRoutes.post('/addNewBatch', asyncHandler(async (req, res, next) => {
+  const { body } = req;
+  const tempStartDate = new Date(body.startDate);
+  body.endDate = addDays(tempStartDate, body.numberOfDays);
+  // TODO body validation
```

Comment:
Validation logic should be offloaded to mongoose. When you call new Batch({...}), with some data, mongoose will throw an appropriate error when the data provided does not match the schema. We can use this instead of writing our own validation logic

Reply:
If some data is missing in the request then the code should be 400 but if there is no server side sanitization the error code will be 500. What do you think?

Reply:
"400 Bad Request" basically means the client (user's browser) made a bad request to the server. This covers both missing and present-but-incorrect/invalid data.

I'm not sure I understand "no server side sanitization" correctly. Could you elaborate?